api/api-plugin-infos-v5/src/test/groovy/com/thoughtworks/go/apiv5/plugininfos/PluginInfosControllerV5Test.groovy

```diff
@@ -0,0 +1,295 @@
+        descriptor.markAsInvalid(new ArrayList<String>(), null)
+        def pluginInfo = new CombinedPluginInfo(new BadPluginInfo(descriptor))
+
+        when(pluginInfoFinder.pluginInfoFor('plugin_id')).thenReturn(null)
+        when(defaultPluginManager.getPluginDescriptorFor('plugin_id')).thenReturn(descriptor)
+
+        getWithApiHeader(controller.controllerPath('/plugin_id'))
+
+        assertThatResponse()
+          .isOk()
+          .hasContentType(controller.mimeType)
+          .hasBodyWithJsonObject(pluginInfo, PluginInfoRepresenter)
+      }
+
+      @Test
+      void 'should return 404 if plugin with unsupported extension type exist'() {
+        def descriptor = new GoPluginDescriptor("plugin_id", "1", new GoPluginDescriptor.About("authorization", "v1", "goVersion1", "go plugin", new GoPluginDescriptor.Vendor("go", "goUrl"), ["os"]), "/home/pluginjar/", null, true)
+        descriptor.markAsInvalid(new ArrayList<String>(), null)
+        def pluginInfo = new CombinedPluginInfo(new PluginInfo(descriptor, "Invalid extension name", null, null))
+
+        when(pluginInfoFinder.pluginInfoFor('plugin_id')).thenReturn(pluginInfo)
+
+        getWithApiHeader(controller.controllerPath('/plugin_id'))
+
+        assertThatResponse()
+          .isNotFound()
+          .hasJsonMessage(notFoundMessage())
+          .hasContentType(controller.mimeType)
+      }
+
+
+      @Test
+      void 'should return 404 if plugin with id does not exist'() {
+
+        when(pluginInfoFinder.pluginInfoFor('plugin_id')).thenReturn(null)
+        when(defaultPluginManager.getPluginDescriptorFor('plugin_id')).thenReturn(null)
+
+        getWithApiHeader(controller.controllerPath('/plugin_id'))
+
+        assertThatResponse()
+          .isNotFound()
+          .hasJsonMessage(notFoundMessage())
+          .hasContentType(controller.mimeType)
+      }
+
+      @Test
+      void 'should return 304 if plugin info is not modified'() {
+        def descriptor = new GoPluginDescriptor("plugin_id", "1", new GoPluginDescriptor.About("authorization", "v1", "goVersion1", "go plugin", new GoPluginDescriptor.Vendor("go", "goUrl"), ["os"]), "/home/pluginjar/", null, true)
+        def pluginInfo = new CombinedPluginInfo(new PluginInfo(descriptor, ExtensionType.AUTHORIZATION.getExtensionType(), null, null))
+
+        when(pluginInfoFinder.pluginInfoFor('plugin_id')).thenReturn(pluginInfo)
+        when(entityHashingService.md5ForEntity(pluginInfo)).thenReturn('md5')
+
+        getWithApiHeader(controller.controllerPath('/plugin_id'), ['if-none-match': '"md5"'])
+
+        assertThatResponse()
+          .isNotModified()
+          .hasContentType(controller.mimeType)
+      }
+
+    }
+  }
+
+  @Nested
+  class Index {
+    @Nested
+    class Security implements SecurityTestTrait {
```

Comment:
Implementing just `SecurityTestTrait` wouldn't do anything.
We need to mention user security, such as `AdminUserSecurity` or `NormalUserSecurity`.

With this we dont have any tests for the controller user permissions.